### PR TITLE
Switch default branch from master to main

### PR DESCRIPTION
**Pull request by tmarkstrum**
## What does this PR do?
Switch default branch from master to main

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -1389,7 +1389,7 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
             # go out of scope, at which point it could otherwise be freed for
             # further reuse by the main stream while the div/reduce_scatter/copy
             # are underway in the post_backward stream. See:
-            # github.com/NVIDIA/apex/blob/master/apex/parallel/distributed.py
+            # github.com/NVIDIA/apex/blob/main/apex/parallel/distributed.py
```
Comment (blefaudeux) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:1392]: false positive ?

File: fairscale/utils/testing.py [hunk 2]
```diff
@@ -571,7 +571,7 @@ def forward(self, *input: Tensor, **kwargs: Any) -> Tensor:
 def get_cycles_per_ms() -> float:
     """Measure and return approximate number of cycles per millisecond for torch.cuda._sleep
 
-    Copied from: github.com/pytorch/pytorch/blob/master/test/test_cuda.py
+    Copied from: github.com/pytorch/pytorch/blob/main/test/test_cuda.py
```
Comment (blefaudeux) [inline fairscale/utils/testing.py:574]: not sure about that actually, false positive ?

**Review by blefaudeux — APPROVED** [review]
LGTM, thanks @tmarkstrum ! Minor question, see comments, maybe a false positive on a nvidia/apex reference

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 1 above)
Comment (tmarkstrum) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py]: Good catch. updated.

File: fairscale/utils/testing.py (hunk 2 above)
Comment (tmarkstrum) [reply fairscale/utils/testing.py]: Yes. They have not updated the branch name yet. I corrected it.

File: README.md [hunk 3]
```diff
@@ -178,3 +178,14 @@ If you use FairScale in your publication, please cite it by using the following
   year =         {2021}
 }
 ```
+
+## FAQ
```
Comment (anj-s) [inline README.md:182]: Should we add a whats new section or a Note section? This FAQ is at the bottom right?

Comment (tmarkstrum) [reply README.md:182]: ok. I think we should continue updating "what new section" as the pytorch/fairseq README.md from now on.

**Pull request merged**